leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/CaliforniumLwM2mRequestSender.java

```diff
@@ -47,6 +51,7 @@
     private final Set<Endpoint> endpoints;
     private final ObservationRegistry observationRegistry;
     private final LwM2mModelProvider modelProvider;
+    private final Map<String/* registration id */, ArrayList<Request> /* pending coap Request */> pendingRequests = new HashMap<>();
```

Comment:
OK, this is "endpoint name" to client, not "register id".
So staying with the "reg-id", the entry in the map should be removed on "cancelPendingRequests", (otherwise this may be something like a memory leak).
Alternatively, we use the endpoint name, copy the request list on cancel and iterate over the copy. I will make a purposal.